NDTensors/src/lib/UnallocatedArrays/src/abstractfill/set_types.jl

```diff
@@ -42,34 +73,9 @@ SetParameters.default_parameter(::Type{<:AbstractFill}, ::Position{3}) = Tuple{}
 
 SetParameters.nparameters(::Type{<:AbstractFill}) = Val(3)
 
-# Set parameter 1
-## Right now using AbstractArray
-## TODO These are more difficult because T is technically defined so need some way to strip T of it {} types
-# function set_parameter(
-#   T::Type{<:AbstractFill{<:Any,<:Any,P3}}, ::Position{1}, P1
-# ) where {P3}
-#   return T{P1,<:Any,P3}
-# end
-# function set_parameter(
-#   T::Type{<:AbstractFill{<:Any,P2,P3}}, ::Position{1}, P1
-# ) where {P2,P3}
-#   return T{P1,P2,P3}
-# end
-# function set_parameter(
-#   T::Type{<:AbstractFill{<:Any,<:Any,P3}}, ::Position{2}, P2
-# ) where {P3}
-#   return T{<:Any,P2,P3}
-# end
-# function set_parameter(
-#   T::Type{<:AbstractFill{P1,<:Any,P3}}, ::Position{2}, P2
-# ) where {P1,P3}
-#   return T
-# end
-# set_parameter(T::Type{<:AbstractFill}, ::Position{3}, P3) = T{<:Any,<:Any,P3}
-# set_parameter(T::Type{<:AbstractFill{P1}}, ::Position{3}, P3) where {P1} = T{<:Any,P3}
-# function set_parameter(T::Type{<:AbstractFill{<:Any,P2}}, ::Position{3}, P3) where {P2}
-#   return T{<:Any,P2,P3}
-# end
-
-# Set parameter 2
-## using AbstractArray
+## This is a helper function which can be renamed.
+## Essentially, because I use AbstractFill I don't know what the name
+## of the object is in `set_parameter`, so I use this to find the 
+## name with out any of the parameters attached.
+strip_parameters(::Type{<:Fill}) = Fill
+strip_parameters(::Type{<:Zeros}) = Zeros
```

Comment:
Good idea to add a function like that. I was thinking we would call it `unspecify_parameters` as the opposite of `specify_parameters` (and in the future you could choose subsets of parameters to unspecify).

Reply:
@mtfishman Should I implement `unspecify_parameters` in the `SetParameters` module or should it just live here for now?

Reply:
Okay I have renamed these these functions and added a set of tests for changing all possible parameters. 
I had two things to bring up: one is that I used `@eval` to write the tests once for both `Fill` and `Zeros` and because of the nested `@eval` issue, I had to put it outside of the random temporary module.
The second is I am wondering if the tests I have would be considered type piracy because I didn't write these AbstractFill object.

Reply:
Unless the tests involve overloading a function I don't see why it would be type piracy, could you leave comments in Github on code you think might be problematic or want reviewed?